packages/utils/src/loader/utils.ts

```diff
@@ -29,38 +17,28 @@ export async function mergeIconProps(
 		const v = additionalProps[p];
 		if (v !== undefined && v !== null) props[p] = v;
 	});
+	// add svg xmlns if missing
+	if (!svg.includes(' xmlns=') && !additionalProps['xmlns']) {
+		additionalProps['xmlns'] = 'http://www.w3.org/2000/svg';
+	}
+	// add xmlns:xlink if xlink present and the xmlns missing
+	if (!svg.includes('xmlns:xlink') && svg.includes('xlink:') && !additionalProps['xmlns:xlink']) {
+		additionalProps['xmlns:xlink'] = 'http://www.w3.org/1999/xlink';
+	}
 	const replacement = svg.startsWith('<svg ') ? '<svg ' : '<svg';
 	return svg.replace(
 		replacement,
-		`${replacement}${Object.keys(props)
-			.map((p) => `${p}="${props[p]}"`)
-			.join(' ')}`
+		`${replacement}${Object.keys(props).map((p) => `${p}="${props[p]}"`).join(' ')}`
 	);
 }
 
-export async function tryInstallPkg(name: string): Promise<void | undefined> {
-	if (pending) {
-		await pending;
-	}
-
-	if (!tasks[name]) {
-		// eslint-disable-next-line no-console
-		console.log(cyan(`Installing ${name}...`));
-		tasks[name] = pending = installPackage(name, {
-			dev: true,
-			preferOffline: true,
-		})
-			.then(() => sleep(300))
-			// eslint-disable-next-line
-			.catch((e: any) => {
-				warnOnce(`Failed to install ${name}`);
-				console.error(e);
-			})
-			.finally(() => {
-				pending = undefined;
-			});
-	}
-
-	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-	return tasks[name]!;
+// https://bl.ocks.org/jennyknuth/222825e315d45a738ed9d6e04c7a88d0
+export function encodeCssSvg(svg: string): string {
```

Comment:
Can you move that to a separate file? I actually want to use this in icon components.

Recently I've been experimenting with SVG 2 animations, they have one big issue: animations don't start until entire document is loaded and something small like failed ad code can break them. However, if used as background/mask, like in `unplugin-icons`, animation starts immediately. This function can be useful to generate URLs.

Reply:
sure

Reply:
maybe another module on `packages/utils/src/svg`, for example `css.ts`?

Reply:
sure, that's perfect place for it.